File: widgetssdk/src/main/java/com/glia/widgets/core/permissions/domain/HasCallNotificationChannelEnabledUseCase.java
package com.glia.widgets.core.permissions.domain;

import com.glia.widgets.core.permissions.PermissionManager;

public class HasCallNotificationChannelEnabledUseCase {
    private final PermissionManager permissionManager;

    public HasCallNotificationChannelEnabledUseCase(PermissionManager permissionManager) {
        this.permissionManager = permissionManager;
    }

    public boolean execute() {
        return this.permissionManager.hasCallNotificationChannelEnabled();
    }
}

File: widgetssdk/src/main/java/com/glia/widgets/core/permissions/domain/HasCallNotificationChannelEnabledUseCase.kt
package com.glia.widgets.core.permissions.domain

import android.content.Context
import com.glia.widgets.core.notification.NotificationFactory
import com.glia.widgets.core.notification.areNotificationsEnabledForChannel

internal class HasCallNotificationChannelEnabledUseCase(private val context: Context) {
    operator fun invoke(): Boolean = context.areNotificationsEnabledForChannel(NotificationFactory.NOTIFICATION_CALL_CHANNEL_ID)
}
